src/bip44.c

```diff
@@ -260,6 +245,76 @@ size_t bip44_printToStr(const bip44_path_t* pathSpec, char* out, size_t outSize)
 	return ptr - out;
 }
 
+
+bip44_path_type_t bip44_classifyPath(const bip44_path_t* pathSpec)
+{
+	if (bip44_hasValidCardanoPoolColdKeyPrefix(pathSpec)) {
+		if (bip44_isValidPoolColdKeyPath(pathSpec)) {
+			return PATH_POOL_COLD_KEY;
+		} else {
+			return PATH_POOL_COLD_KEY_INVALID;
```

Comment:
I don't see any code actually dealing with invalid paths on this granularity - I would just default to `PATH_INVALID` (renaming the PATH_UNKNOWN enum, so it's more accurate)

We can always introduce that kind of granularity if we would actually need it, but for now it seems like unnecessary bloat making this logic more complicated than it needs to be, unless I'm missing something